NorthstarDLL/scripts/client/scriptmodmenu.cpp

```diff
@@ -1,165 +1,108 @@
+	g_pSquirrel<context>->sealstructslot(sqvm, 3);
 
-	return SQRESULT_NULL;
-}
+	// link
+	g_pSquirrel<context>->pushstring(sqvm, mod.DownloadLink.c_str(), -1);
+	g_pSquirrel<context>->sealstructslot(sqvm, 4);
 
-ADD_SQFUNC("void", NSSetModEnabled, "string modName, bool enabled", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
-	const SQBool enabled = g_pSquirrel<context>->getbool(sqvm, 2);
+	// installLocation
+	g_pSquirrel<context>->pushstring(sqvm, mod.m_ModDirectory.generic_string().c_str());
+	g_pSquirrel<context>->sealstructslot(sqvm, 5);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
-		{
-			mod.m_bEnabled = enabled;
-			return SQRESULT_NULL;
-		}
-	}
+	// loadPriority
+	g_pSquirrel<context>->pushinteger(sqvm, mod.LoadPriority);
+	g_pSquirrel<context>->sealstructslot(sqvm, 6);
 
-	return SQRESULT_NULL;
-}
+	// requiredOnClient
+	g_pSquirrel<context>->pushbool(sqvm, mod.RequiredOnClient);
+	g_pSquirrel<context>->sealstructslot(sqvm, 7);
 
-ADD_SQFUNC("string", NSGetModDescriptionByModName, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+	// enabled
+	g_pSquirrel<context>->pushbool(sqvm, mod.m_bEnabled);
+	g_pSquirrel<context>->sealstructslot(sqvm, 8);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
+	// conVars
+	g_pSquirrel<context>->newarray(sqvm, 0);
+	for (ModConVar* conVar : mod.ConVars)
 	{
-		if (!mod.Name.compare(modName))
-		{
-			g_pSquirrel<context>->pushstring(sqvm, mod.Description.c_str());
-			return SQRESULT_NOTNULL;
-		}
+		g_pSquirrel<context>->pushstring(sqvm, conVar->Name.c_str());
+		g_pSquirrel<context>->arrayappend(sqvm, -2);
 	}
+	g_pSquirrel<context>->sealstructslot(sqvm, 9);
 
-	return SQRESULT_NULL;
+	g_pSquirrel<context>->arrayappend(sqvm, -2);
 }
 
-ADD_SQFUNC("string", NSGetModVersionByModName, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
+ADD_SQFUNC("void", NSSetModEnabled, "int index, bool enabled", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
 {
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+	const int index = g_pSquirrel<context>->getinteger(sqvm, 1);
+	const SQBool enabled = g_pSquirrel<context>->getbool(sqvm, 2);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
+	if (index < 0 || index >= g_pModManager->m_LoadedMods.size())
 	{
-		if (!mod.Name.compare(modName))
-		{
-			g_pSquirrel<context>->pushstring(sqvm, mod.Version.c_str());
-			return SQRESULT_NOTNULL;
-		}
+		spdlog::warn("Tried disabling mod at {} but only {} mods are loaded", index, g_pModManager->m_LoadedMods.size());
```

Comment:
Do we know how to push a proper script error? (with the message)

If so we should be doing that here

Reply:
I think I've heard someone say that the function doesn't work but in principle we know.
This should be done in a future PR for all exposed functions imo

Reply:
@uniboi do we know?

Reply:
AFAIK we do not know how to get it working right now. I can try some things but would prefer it in a future PR as said above